lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/BuildJsonHttpDeserializer.kt

```diff
@@ -52,6 +52,8 @@ internal fun <T> buildJsonHttpDeserializer(
                 response.body,
             )
 
+        // TODO: In case of batch request, exception should not be thrown while checking
+        // each single response. Instead, it should be thrown when accessing specific call with an error.
```

Comment:
what do you mean by 'checking'?
I generally agree that exception should be thrown when accessing specific call with an error, unless there was some network error/node error - then we should throw it straight away (eg when node returned malformed response, or is unreachable)

Reply:
To be precisely, by checking I meant deserializing each response. I updated todo comment to make it more clear.